src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +13,166 @@ class MatchSamplingRate(Transform):
-    | -------------- | --------- | -------------- | --------- | -------- |
-    | [0, 1, 2]      | [2, 1, 7] | [0, 2]         | [10, 20]  | 1        |
-    | [0, 1, 2]      | [4, 1, 0] | [0, 2]         | [20, 40]  | 2        |
+    | feature_a                         | feature_b                          | const |
+    | ---                               | ---                                | ---   |
+    | list[struct[datetime[us],struct[]]| list[struct[datetime[us],struct[]] | int   |
+    | ----------------------------------| -----------------------------------| ----- |
+    | [{2024-06-25 12:26:01.0,{1.2}},   | [{2024-06-25 12:26:00.0,{1.0}},    | 1     |
+    |  {2024-06-25 12:26:02.0,{2.4}},   |  {2024-06-25 12:26:05.0,{2.0}}]    |       |
+    |  {2024-06-25 12:26:03.0,{3.6}},   |                                    |       |
+    |  {2024-06-25 12:26:04.0,{4.8}}]   |                                    |       |
 
     The following transform can be used to match the sampling rate
-    of the time series described by `time_feature_b` and
-    `feature_b` to the sampling rate of the time series described by
-    `time_feature_a` and `feature_a`.
+    of the time series `feature_b` to the sampling rate
+    of the time series `feature_a`.
 
     ```
-        transforms:
-            - classpath: agenc.transforms.MatchSamplingRate
-                arguments:
-                    reference_timestamps: time_feature_a
-                    feature_columns_with_timestamps:
-                        feature_b: time_feature_b
-            - ...
+        ...
+        environment.load()
+        data = environment.get_data()
+        transform = MatchSamplingRate(
+            reference_feature="feature_a",
+            feature_columns={
+                "feature_b": "linear",
+            },
+        )
+        transformed_data = transform.transform(data)
+        ...
     ```
 
     The resulting Dataframe after the transform is:
 
-    | time_feature_a | feature_a   | time_feature_b | feature_b    | constant |
-    | -------------- | ----------- | -------------- | ------------ | -------- |
-    | [0, 1, 2]      | [2, 1, 7]   | [0, 1, 2]       | [10, 15, 20]    | 1    |
-    | [0, 1, 2]      | [4, 1, 0]   | [0, 1, 2]       | [20, 30, 40]    | 2    |
+    | feature_a                          | feature_b                          | const |
+    | ---                                | ---                                | ---   |
+    | list[struct[datetime[us],struct[]] | list[struct[datetime[us],struct[]] | int   |
+    | -----------------------------------| -----------------------------------| ----- |
+    | [{2024-06-25 12:26:01.0,{1.2}},    | [{2024-06-25 12:26:00.0,{1.0}},    | 1     |
+    |  {2024-06-25 12:26:02.0,{2.4}},    |  {2024-06-25 12:26:01.0,{1.2}},    |       |
+    |  {2024-06-25 12:26:03.0,{3.6}},    |  {2024-06-25 12:26:02.0,{2.4}},    |       |
+    |  {2024-06-25 12:26:04.0,{4.8}}]    |  {2024-06-25 12:26:03.0,{3.6}}]    |       |
```

Comment:
huh?! why is the time series not the same after interpolation? I'd assume that `feature_a` and `feature_b` have the same `datetime`s after interpolation?!

Reply:
they do have the same `datetime`s. I fixed the error in the docstring